src/lib/components/Toggle.react.js

```diff
@@ -0,0 +1,108 @@
+import React, { Component } from 'react';
+import { withStyles } from '@material-ui/core/styles';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Switch from '@material-ui/core/Switch';
+import Grid from '@material-ui/core/Grid';
+import Typography from '@material-ui/core/Typography';
+
+type Props = {
+  /** CSS class name of the root element */
+  className?: string,
+  /** The classname of the root of the toggle switch */
+  classNameRoot?: string,
+  /** Whether the toggle is disabled (true) or not (false) */
+  disabled?: boolean,
+  /** Toggle ID */
+  id: string,
+  /** The label for the toggle */
+  label?: string,
+  /** If using a single label, its position can be: "top", "start", "bottom", or "end" */
+  labelPlacement?: string,
+  /** The space between the label(s) and toggle */
+  labelSpacing?: int,
+  /**
+   * A second label for the toggle. If this is used, the labelPlacement value is ignored, and the
+   * secondaryLabel will be positioned to the right of the toggle, and the first label to the left
+   */
+  secondaryLabel?: string,
+  /** Dash callback to update props on the server */
+  setProps?: (props: { modal?: boolean, open?: boolean }) => void,
+  /** Whether toggle is on (true) or off (false) */
+  toggled: boolean,
+};
+
+type State = {
+  toggled: boolean,
+};
+
+const defaultProps = {
+  className: '',
+  classNameRoot: '',
+  disabled: false,
+  labelPlacement: "right",
+  label: '',
+  labelSpacing: 1,
+  secondaryLabel: '',
+  setProps: () => {},
+}
+
+export default class Toggle extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {toggled: props.toggled};
+  }
+
+  handleChange = (event) => {
+    this.setState({toggled: event.target.checked });
+
+    if (typeof this.props.setProps === 'function') {
+      this.props.setProps({toggled: event.target.checked});
+    }
+  };
+
+  render() {
+    const {
+      id,
+      className,
+      classNameRoot,
+      disabled,
+      label,
+      secondaryLabel,
+      labelPlacement} = this.props;
+
+    return secondaryLabel ? (
+      <div id={id} className={className}>
+        <Typography component="div">
+          <Grid component="label" container alignItems="center" spacing={1}>
+            <Grid item>{label}</Grid>
+            <Grid item>
+              <Switch
+                checked={this.state.toggled}
+                onChange={this.handleChange}
+                disabled={disabled}
+                classes={{root: classNameRoot}}
```

Comment:
So I can't pass in an arbitrary `classes` argument?

Reply:
I don't think we could pass one in that's entirely arbitrary. I do think we could make this more flexible by letting the user decide which keys they assign classnames to, but that would require that the user know the appropriate [CSS rule names ](https://material-ui.com/api/switch/#css).

Reply:
Why not?

I'd prefer to pass in `classes=dict(root='my-root', label='my-label-class')` from Dash since that's the way MUI does it now.

Reply:
Label is not a recognized CSS rule name for this component. We could add an optional prop for each of the valid rule names, though.

Reply:
My bad, replace `label` with `colorPrimary` then as a key to match it to the `Switch` classes options. I'm just saying I'd prefer to support the generic `classes` prop as a dictionary (Object) than to have numerous additional props that may change at any time if MUI changes the prop names/adds more.

e.g.,

```python
classes=dict(
    root='my-root',
    colorPrimary='color-primary-class',
    input='input-class',
    thumb='thumb-class',
)
```